src/Tools/BuildValidator/Extensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace BuildValidator
+{
+    internal static class Extensions
+    {
+#if !NETCOREAPP
+        internal static void Write(this FileStream stream, ReadOnlySpan<byte> span)
+        {
+            for (int i = 0; i < span.Length; i++)
+            {
+                stream.WriteByte(span[i]);
+            }
+        }
+#endif
+
+        internal static MetadataReader GetEmbeddedPdbMetadataReader(this PEReader peReader)
+        {
+            var entry = peReader.ReadDebugDirectory().Single(x => x.Type == DebugDirectoryEntryType.EmbeddedPortablePdb);
+            var provider = peReader.ReadEmbeddedPortablePdbDebugDirectoryData(entry);
```

Comment:
Is this change based on the rationale that we only care about comparing the PDBs if they are embedded, and if the pdb is instead produced as a separate file, we don't care whether it's the same as the original?

Reply:
This change was motivated by the idea that this tool will eventually be a library and the library shouldn't be going to the disk for basic information like getting a `MetadataReader`. There are essentially two cases for the rebuild scenario when it comes to PDB production: 

- The PDB is embedded in which case this method can get a `MetadatReader` to it without going to disk 
- The PDB is not embedded in which case we have a `Stream` from which we can produce a `MetadataReader`

This is solving the first case.